### PR TITLE
Support to honour custom retry gaps for attach/detach

### PR DESCRIPTION
**Pull request by sameshai**
Below is the custom retry logic

```
MaxVPCRetryAttempt    int    `toml:"max_vpc_retry_attempt,omitempty" envconfig:"MAX_VPC_RETRY_ATTEMPT"`
MinVPCRetryGap        int    `toml:"min_vpc_retry_gap,omitempty" envconfig:"MIN_VPC_RETRY_INTERVAL"`
MinVPCRetryGapAttempt int    `toml:"min_vpc_retry_gap_attempt,omitempty" envconfig:"MIN_VPC_RETRY_INTERVAL_ATTEMPT"`

```
Default MaxVPCRetryAttempt = 46  sec, MinVPCRetryGap = 3 sec , MinVPCRetryGapAttempt = 3 sec
1.) Honour the MinVPCRetryGap only if it is greater than 3 and less than 10 sec
2.) Honour the MinVPCRetryGapAttempt only if it is greater than 0
3.) Honour the MaxVPCRetryAttempt only if it is greater than 46 ( 7 mins default)

**Flow** 

1.) First attempt is immediately after attach was done.
2.) MinVPCRetryGapAttempts will be done with interval MinVPCRetryGap
3.) 2*MinVPCRetryGapAttempts will be done with interval of MinVPCRetryGap*2 ( it will default to 10 sec if it more than 10)
4.) Remaining attempts will be done with interval of 10 secs

### CODE REVIEW

File: block/provider/provider.go [hunk 1]
```diff
@@ -226,6 +226,21 @@ func (vpcp *VPCBlockProvider) OpenSession(ctx context.Context, contextCredential
...
+	if vpcp.Config.VPCConfig.MinVPCRetryGap > ConstMinVPCRetryGap && vpcp.Config.VPCConfig.MinVPCRetryGap < ConstantRetryGap {
+		minVPCRetryGap = vpcp.Config.VPCConfig.MinVPCRetryGap
+	}
+
+	if vpcp.Config.VPCConfig.MinVPCRetryGapAttempt > 0 {
+		minVPCRetryGapAttempt = vpcp.Config.VPCConfig.MinVPCRetryGapAttempt
+	}
+
+	if vpcp.Config.VPCConfig.MaxVPCRetryAttempt > ConstMaxVPCRetryAttempt {
+		maxVPCRetryAttempt = vpcp.Config.VPCConfig.MaxVPCRetryAttempt
```
Comment (skorati) [inline block/provider/provider.go:239]: maxVPCRetryAttempt -> This should be dynamically calculated based on minVPCRetryGapAttempt, minVPCRetryGap each time.

File: block/provider/util.go [hunk 2]
```diff
@@ -218,6 +235,45 @@ func (fRetry *FlexyRetry) FlexyRetryWithConstGap(logger *zap.Logger, funcToRetry
 	return err
 }
 
+// FlexyRetryWithCustomGap ...
```
Comment (skorati) [inline block/provider/util.go:238]: Can you please add comments for readability.

**Review by skorati — COMMENTED** [review]
Added comments

**Review by skorati — APPROVED** [review]
lgtm

**Review by prankulmahajan — APPROVED** [review]
Looks good

**Comment by sameshai** [comment]
Merging this refer travis for unitest results  TestFlexyRetryWithCustomGap

https://app.travis-ci.com/github/IBM/ibmcloud-volume-vpc/jobs/583049596#L306

**Review by skorati — APPROVED** [review]

**Review by ambiknai — APPROVED** [review]

**Pull request merged**